tree/treeplayer/test/treeprocs/treeprocessors.cxx

```diff
@@ -56,6 +59,42 @@ void DeleteFiles(const std::vector<std::string> &filenames)
       gSystem->Unlink(f.c_str());
 }
 
+class FilesRAII {
+   std::vector<std::string> fFileNames;
+
+public:
+   FilesRAII(const std::vector<std::string> &treenames, const std::vector<std::string> &filenames, int nEvts = 10)
+      : fFileNames(filenames)
+   {
+      WriteFiles(treenames, filenames, nEvts);
+   }
+   ~FilesRAII() { DeleteFiles(fFileNames); }
+};
+
+// See issue #15425
+TEST(TreeProcessorMP, moreWorkersThanEvents)
+{
+   auto func = [](TTreeReader &r){
+      int n=0;
+      while(r.Next()) n++;
+      auto par = new TParameter<int>("n", n);
```

Comment:
I've never seen this class used, is it safe to allocate it without deleting it? Whose taking responsibility?

Reply:
That is allocated within a subprocess which is then killed. The histogram returned by `TProcessorMP::Process` should be deleted though. Thanks.